### DISCUSSION TITLE
Weird filter methods

### DISCUSSION THREAD

**Discussion by kxzo** [prisma/prisma#15198] [Q&A]
Hi, I try to filter the users that only users that neither are you, nor your friends are left.

However, I just get the user that isn't you, but the other filter with the friendSender... doesn't work (just the id).

Schema:
```prisma
model User { 
   id         String   @id @default(auto()) @map("_id") @db.ObjectId 
   email      String   @unique(map: "userEmail_1") 
   password   String 
   picture    String? 
   token      String   @unique(map: "userToken_1") 
   userName   String   @unique(map: "userName_1") 
   likedGames String[] @default([]) @db.ObjectId 
  
   friendReceiver FriendPair[] @relation("receiver") 
   friendSender   FriendPair[] @relation("sender") 
 } 
  
 model FriendPair { 
   id           String   @id @default(auto()) @map("_id") @db.ObjectId 
   receiverName String 
   receiverId   String 
   receiver     User     @relation("receiver", fields: [receiverId], references: [id]) 
   senderName   String 
   senderId     String 
   sender       User     @relation("sender", fields: [senderId], references: [id]) 
   messages     Json[]   @default([]) 
   sent         DateTime @default(now()) 
   status       String   @default("pending") 
 }
```

The recommendation function that won't work properly:

```js
const recUsers = await prisma.user.findMany({ 
       where: { 
         id: { not: user.id }, 
         OR: [ 
           { 
             friendSender: { 
               some: { 
                 receiverId: { 
                   not: user.id, 
                 }, 
                 senderId: { 
                   not: user.id, 
                 }, 
               }, 
             }, 
           }, 
           { 
             friendReceiver: { 
               every: { 
                 receiverId: { 
                   not: user.id, 
                 }, 
                 senderId: { 
                   not: user.id, 
                 }, 
               }, 
             }, 
           }, 
         ], 
       }, 
       orderBy: { 
         id: 'desc', 
       }, 
       take: 100, 
     });
```

**Comment by pantharshit00** [comment]
Hey! Can you please explain a bit more on what you are trying to filter? I am unable to understand it completely. Maybe give an example with some sample data.

**Reply by kxzo** [reply]
Sure, so I actually want to have a "friendPair" that contains the chat messages, the status of the request and should contain the user that send the request (for friend) and the user that is the receiver of the request (friend request).

**So I want to connect the FriendPair with two Users but I don't know how.**
**In the end, there should only be like one relation in the User model for the FriendPair model.**

**Reply by kxzo** [reply]
I would really appreciate any kind of 'helpful' response, please, I am quite desperate, not getting the bugs fixed for weeks :c

**Reply by nurul3101** [reply]
From the first message, if I understand correctly, you only want the users to whom a pair request is not sent or received. Am I correct?

**Reply by kxzo** [reply]
> From the first message, if I understand correctly, you only want the users to whom a pair request is not sent or received. Am I correct?

You answered a bit late lmao. I managed to solve the issue after some weeks of tweaking.

**Reply by nurul3101** [reply]
I am glad to hear that you were able to solve it 😄
